catkin_ws/src/image_rotation/src/processor_node.cpp

```diff
@@ -0,0 +1,27 @@
+///
+/// @file
+/// @brief Implementation of the ProcessorNode class
+///
+
+#include "processor_node.h"
+
+ProcessorNode::ProcessorNode(const std::string& topic)
+:image_transport_handle_(node_handle_),
+ topic_(topic) 
+{
+    publisher_ = image_transport_handle_.advertise((topic_+"_rotated"),10);
+    subscriber_ = image_transport_handle_.subscribe(topic_,10,&ProcessorNode::ImageRotationCallBack,this);
+}
+
+void ProcessorNode::ImageRotationCallBack(const sensor_msgs::ImageConstPtr& msg)
+{
+    const float rotation_angle = 180.0f;
```

Comment:
i suppose this could be a const class member to make this generic for any angle of rotation  .(optional ).